frontend/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -57,10 +61,12 @@ function renderFigure({
   const isFullScreen = (): boolean => !!height
 
   const theme: Theme = useTheme()
+  let updatedSpec = figure.spec
 
   const generateSpec = (figure: FigureProto): any => {
-    const spec = JSON.parse(figure.spec)
+    updatedSpec = replaceTemporaryColors(updatedSpec, theme, element.theme)
```

Comment:
What is the reason to initialize the variable outside? Wouldn't a one-liner here also work?:

```typescript
const updatedSpec = replaceTemporaryColors(figure.spec, theme, element.theme)
```